src/txn/txn_rollback_to_stable.c

```diff
@@ -8,8 +8,9 @@
 
 #include "wt_internal.h"
 
-#define WT_CHECK_RECOVERY_FLAG_TXNID(session, txnid) \
-    (F_ISSET(S2C(session), WT_CONN_RECOVERING) && (txnid) >= S2C(session)->recovery_ckpt_snap_min)
+#define WT_CHECK_RECOVERY_FLAG_TXNID(session, txnid)                                           \
+    (F_ISSET(S2C(session), WT_CONN_RECOVERING) && S2C(session)->recovery_ckpt_snap_min != 0 && \
+      (txnid) >= S2C(session)->recovery_ckpt_snap_min)
```

Comment:
This change does not seem version related. Is it necessary? Should it be its own ticket?

Reply:
Yes, this change is not version related, but having this check of whether there is a checkpoint snapshot or not can improve the RTS performance if the table has aggregated time window containing transaction id. Otherwise, RTS opens all the files that have aggregated transaction IDs and does nothing.

Reply:
Thanks.